cadquery/occ_impl/solver.py

```diff
@@ -228,34 +226,50 @@ def jac(x):
 
                             if k1 not in self.locked:
                                 tmp1 = pnt_pln_cost(m1, m2, t1j, t2, d)
-                                rv[k1 * NDOF + j] += (tmp1 - tmp) / DIFF_EPS
+                                rv[k1 * NDOF + j] += 2 * tmp * (tmp1 - tmp) / DIFF_EPS
 
                             if k2 not in self.locked:
                                 tmp2 = pnt_pln_cost(m1, m2, t1, t2j, d)
-                                rv[k2 * NDOF + j] += (tmp2 - tmp) / DIFF_EPS
+                                rv[k2 * NDOF + j] += 2 * tmp * (tmp2 - tmp) / DIFF_EPS
                     else:
                         raise NotImplementedError(f"{m1,m2}")
 
             return rv
 
         return f, jac
 
-    def solve(self) -> List[Location]:
+    def solve(self) -> Tuple[List[Location], Dict[str, Any]]:
 
         x0 = array([el for el in self.entities]).ravel()
         f, jac = self._cost()
 
-        res = minimize(
-            f,
-            x0,
-            jac=jac,
-            method="BFGS",
-            options=dict(disp=True, gtol=1e-12, maxiter=1000),
-        )
+        def func(x, grad):
+
+            if grad.size > 0:
+                grad[:] = jac(x)
+
+            return f(x)
 
-        x = res.x
+        opt = nlopt.opt(nlopt.LD_CCSAQ, len(x0))
+        opt.set_min_objective(func)
 
-        return [
-            Location(self._build_transform(*x[NDOF * i : NDOF * (i + 1)]))
-            for i in range(self.ne)
-        ]
+        opt.set_ftol_abs(0)
+        opt.set_ftol_rel(0)
+        opt.set_xtol_rel(TOL)
+        opt.set_xtol_abs(0)
+        opt.set_maxeval(MAXITER)
+
+        x = opt.optimize(x0)
+        result = {
+            "cost": opt.last_optimum_value(),
+            "iters": opt.get_numevals(),
+            "status": opt.last_optimize_result(),
+        }
+
+        return (
+            [
+                Location(self._build_transform(*x[NDOF * i : NDOF * (i + 1)]))
+                for i in range(self.ne)
+            ],
+            result,
+        )
```

Comment:
Just a neater style that still passes black:
```suggestion
        locs = [
            Location(self._build_transform(*x[NDOF * i : NDOF * (i + 1)]))
            for i in range(self.ne)
        ]
        return locs, result
```